### ISSUE TITLE
Angular.io Website SSL Certificate Problem

### ISSUE THREAD

**Issue by nickhod** [angular/angular.io#1875]
The angular.io website does not work with ESET NOD32, I believe there is a certificate problem.
The certificate has many spam sites in the SAN field.

In NOD32 I have to "Disable HTTPS checking" to view the site or it is blocked.

This happens on no other sites that I visit regularly.

**Comment by ocombe** [comment]
I have the same issue, it's related to the certificate for firebase (which is the same for angular.io), you probably have the same issue with firebaseapp.com
But I think the problems is with Eset, not really Google

**Comment by nickhod** [comment]
I think it's related to the number of 'questionable quality' sites listed on certificate.

https://www.sslshopper.com/ssl-checker.html#hostname=angular.io

Angular should get their own certificate to sidestep the problem.
Wasted two hours trying to figure out why the site wouldn't load. It used to work fine.

**Comment by ocombe** [comment]
Yes, that's a good idea, and for firebase as well, it should be an official and exclusive google certificate

**Comment by akarel** [comment]
Same issue here. ESET NOD32 was the culprit.

**Comment by asaf050** [comment]
Turn off SSL filtering solved it
http://support.eset.com/kb3126/?locale=en_US

**Comment by jhjdev** [comment]
I also had the same issue. Angular.io seems is so far the only site I've encountered this. I can access it on my mobile on the same network without any problems. 
When I disable the HTTPS checking in ESET NOD32 I can access the site without a problem.

**Comment by jhjdev** [comment]
It's also possible to load the website under ESET NOD32 by disabling HTTPS scanning (guessing it's almost the same thing?)
![udklip](https://cloud.githubusercontent.com/assets/968553/17059034/864d2e7a-5023-11e6-8bf5-3f2c13a99a11.PNG)

**Comment by kentweigel** [comment]
For those who would rather only allow access for the certificate used by angular.io, you can 
Note: this refers to NOD32 ANTIVIRUS 9
Click "Setup"
Click "Internet Protection"
Click the gear to the right of "Web access protection"
Click "WEB AND EMAIL"
Expand "SSL/TLS"
Click the "Edit" link next to "List of known certificates"
Click "Add"
Click "URL"
Enter https://angular.io
Choose "Auto" for "Access action"
Choose "Ignore" for "Scan action"
Click "OK" 3 times to close 3 dialog boxes (one click each)
Try again to browse to https://angular.io
A dialog will come up about encrypted network traffic. Select "Remember Action for this certificate" and then click "Allow".

**Comment by ShaneCourtrille** [comment]
It appears this problem can be extended to those using FortiGate firewalls so a new shiny SSL certificate would definitely be a nice thing.

**Comment by ocombe** [comment]
You can build the repo locally: https://github.com/angular/angular.io

**Comment by ShaneCourtrille** [comment]
I'm actually trying that @ocombe but when I do my gulp server-and-sync the Chrome tab that is trying to load it just spins for forever

**Comment by marcusreese** [comment]
Today I can't access angular.io using Chrome. I get this message:
This site can’t provide a secure connection
angular.io didn’t accept your login certificate, or your login certificate may have expired.
Try contacting the system admin.
ERR_BAD_SSL_CLIENT_AUTH_CERT

**Comment by jeffbcross** [comment]
@marcusreese that's unusual. Is it still occurring, or was it temporary?

**Comment by ocombe** [comment]
it's still occurring to all of us who use these firewalls / antivirus softwares

**Comment by ShaneCourtrille** [comment]
I had our security people do some config changes to allow access but the firebase CDN IP changed yesterday so that broke that.  It's looking like we either need to spend time/effort replicating the documentation locally or find another framework to use.  I'm wondering how many other Enterprises (who would be the more likely consumers of Fortigate firewalls) are going to have the same choice because of one silly SSL certificate.

**Comment by ocombe** [comment]
@jeffbcross I can show you the problem at angular connect next week if you want

**Comment by fletort** [comment]
Same problem for me.... and impossible to modify the FortiGate Firewall of the place where is my entreprise..... and we are working with angular.... so it's very very difficult to can get access to documentation :-(.

**Comment by ocombe** [comment]
To all of you who struggle to access the documentation, it is also available on devdocs: http://devdocs.io/angular~2.0_typescript/

**Comment by ShaneCourtrille** [comment]
I'm also in contact with Firebase to see if they can fix anything on there side as this issue impacts all the sites on that SSL certificate.

**Comment by ShaneCourtrille** [comment]
@ocombe Images aren't working as per http://devdocs.io/angular~2.0_typescript/cookbook/component-communication

Not sure if that's something that can be resolved or where I'd report it but still a nice workaround for now.

**Comment by BLITZandKILL** [comment]
Same issue here, I as well am behind a FortiGate firewall

**Comment by drhc** [comment]
Same issue here as well - behind a FortiGate firewall.

**Comment by urbanhusky** [comment]
Same issue. Both Bitdefender and ESET won't allow for a connection to be made.
The workaround in https://github.com/angular/angular.io/issues/1875#issuecomment-234679100 can not be applied because the options in the comment are no longer available.

This is critical for us, because we need to access the site.

**Comment by naomiblack** [comment]
sorry, folks -- I thought we had this fixed just before angular connect. continuing to investigate.

issue reports are very helpful.

**Comment by jeffswitzer** [comment]
A temp work around allowing bitdefender users to access the site is to go to web protection -> settings and uncheck scan ssl. I tried adding angular.io to the whitelist but that didn't work.

**Comment by mbleigh** [comment]
Hey folks -- Firebase Hosting team member here. We're looking into why this might be and what we can do to mitigate. Not sure when we'll be able to find a solution, but we're aware of the issue.

**Comment by janoc** [comment]
This is very much a Fortinet/FortiGate problem - if their appliance is configured for deep inspection, it will try to substitute a forged self-signed certificate (essentially doing a man-in-the-middle attack). Angular.io is using HSTS, so the connection will fail. 

This is a known problem with FortiGate:
http://kb.fortinet.com/kb/documentLink.do?externalID=FD37211

Unfortunately, many of us don't have the choice about having our connection spied on, so we are pretty much screwed :(

**Comment by deltaag** [comment]
I had same problem using Bitdefender Internet Security 2016.

The solution which looks like is working for now is:
Bitdefender -> Modules -> Web Protection -> Whitelist -> Add URL angular.io
I had to stop Scan SSL to get this working, but starting it again didn't cause any more issues.

**Comment by urbanhusky** [comment]
I wouldn't call disabling your antivirus a _solution_

**Comment by mbleigh** [comment]
If HSTS is the issue, I'm not sure what the workaround can be, because we're very opinionated about using HSTS and doing so is deeply ingrained in Firebase Hosting infrastructure. :disappointed:

**Comment by janoc** [comment]
@mbleigh I think nobody can blame you for that. I have managed to get my case solved by hounding the local IT staff to exclude your servers from the FortiGate packet inspection - we are getting our Internet from a municipal connection which has a contract with Fortinet. 

Hopefully I will annoy them sufficiently often (because it isn't only angular.io that doesn't work) that they will disable this spying nonsense for us completely than to have to deal with me again. 

I think the best you can do is to keep this page alive and well document the problem, so that you don't get blamed for problems caused by corporate spyware.

**Comment by nickhod** [comment]
Don't host with Firebase seems like a simple solution to me. It's a simple website with no login; insisting on HSTS at the expense of annoying every user with an SSL inspecting antivirus is a poor trade off.

**Comment by janoc** [comment]
IMO, SSL breaking antiviruses and similar tools that install bogus certificates in your system or try to do man-in-the-middle attacks to allow data inspection are a much bigger problem. It is not long time ago some major ones were found full of security holes and these bogus certificates are a wonderful backdoor ready for exploitation by malware. 

The antivirus vendors should rather work with browser vendors to obtain hooks where to connect the scanning machinery, if the user so desires.

**Comment by urbanhusky** [comment]
@janoc the issue with some bad implementations were that the root certificate was shared on all devices. Modern solutions generate a per-device unique root certificate. That's pretty much impossible to MITM unless you're already on the machine (in which case, all hope is lost anyway). They also forward all the details about the certificate and don't just sign invalid certificates.

**Comment by janoc** [comment]
@urbanhusky Right - and you are left to trust the vendor to do the right thing - like not to siphon off the browsing data for market research and malware sample collection. Do you really have so much trust in a random self-interested third-party with e.g. your banking login? I certainly don't.

In. the Fortinet case, their system generates a completely bogus, self signed certificate, with their "solution" to these problems being a forced installation of their own bogus CA into your browser/certificate store to disable the invalid certificate warnings/HSTS errors. And the company is completely opaque about their policies, who can see the data, etc. Not exactly something I would trust with my data, especially as the company is in the US where the various privacy protection laws are very loose.

**Comment by jeffbcross** [comment]
@janoc are you also not able to visit angularjs.org? We have HSTS configured there, but we have a wildcard certificate dedicated to *.angularjs.org, not shared with any other domains.

@nickhod we have HSTS configured on all Angular domains, even ones not hosted on Firebase.

**Comment by janoc** [comment]
@jeffbcross I think Angularjs.org was working for me, only angular.io hasn't. It is possible that the .org domain is whitelisted by Fortinet and the .io domain is treated as "suspicious" by default.

**Comment by jeffbcross** [comment]
@janoc so does Fortigate only do the MITM for domains it deems suspicious, even if deep inspection is enabled? 

FWIW I sent a message to Fortinet last week through their contact form asking them to whitelist angular.io. If anyone has a better contact, please share.

**Comment by janoc** [comment]
@jeffbcross No they do it on all domains, except what is in the whitelist, I believe. I think they exclude by default sites they class in categories like banking (I believe 3 categories are whitelisted by default). 

The lists are secret, obviously.

**Comment by nickhod** [comment]
@jeffbcross I just re-enabled HTTPS checking on NOD32 and angular.io actually works now. They must have whitelisted it in the past month or so.

I think you're correct that the alternative names on the cert was the issue not HSTS.

**Comment by jeffbcross** [comment]
@nickhod great! 

@ocombe @akarel @asaf050 @jhjdev can you verify that this is resolved for you on NOD32?

**Comment by ocombe** [comment]
@jeffbcross It works with Eset Smart Security (nod32) now :) !!

**Comment by jeffbcross** [comment]
Alright, so at least the original program is fixed, though Fortigate and others are presumably still affected. We'll continue exploring ways to solve this in the meantime.

**Comment by urbanhusky** [comment]
It doesn't work for me in ESET Smart Security (which is just NOD32 with additional functionality) when re-enabling HTTPS scanning. I haven't whitelisted anything.

Firefox:
```
Secure Connection Failed

The connection to the server was reset while the page was loading.

    The page you are trying to view cannot be shown because the authenticity of the received data could not be verified.
    Please contact the web site owners to inform them of this problem.
```

Chrome:
```
This site can’t be reached

The connection was reset.
Try:
Checking the connection
Checking the proxy and the firewall
Running Windows Network Diagnostics
ERR_CONNECTION_RESET
```
I have not restarted yet. Smart Security is on the latest version and all definitions are up-to-date.

**Comment by ocombe** [comment]
I just tested at work and it still doesn't work here, I wonder if I have a different version of Smart Security from my home computer. My version at work is 9.0.402.1

**Comment by tangorri** [comment]
Really lose time with issue. I disabled SSL checking in NOD32, loaded successfully angular.io, now I can load it again withj SSL checking enabled back.

**Comment by deltaag** [comment]
It works for a bit after re-enabling SLL scan, then stops. It might be browser cache or something. Definitely not resolved using Bitdefender.

**Comment by MichalFoksa** [comment]
Same problem here with [https://material.angularjs.org](https://material.angularjs.org ) and ESET Smart Security. 

Above workaround by @kentweigel solved the issue, just add `material.angularjs.org` domain instead of just (or on top of) `angularjs.org` while both of them have different certificate.

**Comment by ocombe** [comment]
I have the same version at home than at work and it works at home but not at work... weird

**Comment by syrusstk** [comment]
angular.io and firebase,io
Chrome: https error, cannot bypass, cannot access websites
Opera - https error, cannot bypass, cannot access websites
Firefox - ok
Edge - ok
Safari - ok

I tried everything possible to bypass in chrome this error and nothing worked, I hate that chrome imposes to not give me an option to bypass their error screen.

Have no firewall enabled, no antivirus, reinstalled chrome a few times with full cleanup,

**Comment by jlsjonas** [comment]
@syrusstk I can confirm it's revoked on safari aswell,
![Screenshot Unable to establish secure connection on Safari](https://i.imgur.com/1Ax5RVR.png)

**Comment by marcin-piekarski** [comment]
Using Bitdefender Total Security 2015 here and I'm getting the same issue as everyone else.

Temporary work around > Turn off Scan SSL under Protection > Web Protection.

Would much rather not have to turn off SSL protection to access Angular.io.

**Comment by nicolasxu** [comment]
I am using Mac, what should I do on Mac to solve this problem?

**Comment by jlsjonas** [comment]
@nicolasxu it's still working on firefox so... I'd suggest to use firefox to open angular.io for now, until they fix the underlying issue

**Comment by nicolasxu** [comment]
@jlsjonas tried firefox, safari. All not working.
<img width="999" alt="screen shot 2016-10-15 at 10 46 12 am" src="https://cloud.githubusercontent.com/assets/1337718/19410893/e57ac270-92c4-11e6-89f1-ac35dca3ef4b.png">

**Comment by jlsjonas** [comment]
@nicolasxu hmm... odd... (though... expected)

for some reason my firefox still thinks it's a valid certificate, all other browsers don't though
![screenshot](https://i.imgur.com/iwQQF1C.png)

I do notice their certificate is only valid from 5october 2016, maybe check your date/time?

**Comment by nicolasxu** [comment]
My clock is auto synced with Apple time server. I tried everything already, still not working. The error msg is "angular.io didn’t accept your login certificate, or your login certificate may have expired."

Why login certificate? I am not trying to login anyting.

**Comment by morriq** [comment]
+1. macbook. won't work on any browser.
halp

**Comment by Martinspire** [comment]
I would like to suggest that going to whitelist every antivirus might not be the road to go. Since Angular will be used in various networks, it should always be available for developers to look at the API and the site. No matter what security suite they are using. And we know that most companies aren't really fast in updating their software, even for stuff like this. And i know for sure that most developers won't be able to change their antivirus settings to disable anything.

I can understand the want for HSTS and other stuff, but perhaps thats overkill for a site thats simply there to view information. What attacks are you trying to prevent there? And it seems that the time just isn't right for using HSTS right now (or whatever is still causing this issue)

If you want people to stop using your new fancy library, you should probably at least make them able to view the API as it is very hard to work without it if you are just starting.

**Comment by ocombe** [comment]
Wouldn't it be simpler to just change the https certificate ? It's not like it costs a lot (especially for Google), you can even get one for free now...

**Comment by doplumi** [comment]
I am on macOS, I don't have any antivirus/firewall installed and I still can't visit the site from Chrome/Chrome Canary/Safari/Safari Technology Preview. Only Firefox is OK with the site for some reason.

Is anybody else here in the same (no antivirus/firewall and still not able to see the site) condition?

**Comment by morriq** [comment]
@doplumi as above..

**Comment by syrusstk** [comment]
@doplumi. Had the same problem only firefox worked and only one pc in my home network had the problem, the next morning everything worked normaly for some reason firefox is the exception

**Comment by Martinspire** [comment]
It seems that https://elements.polymer-project.org got the same issue as this? Come on Google, get your SSL in order!!

**Comment by sebhildebrandt** [comment]
Having the same problem. Working on OS X, not behind a firewall. Can't open angular.io - except on firefox. Any ideas what to do?

**Comment by doplumi** [comment]
Yesterday, on my Mac, it started working again, no action taken.

Il mer 19 ott 2016, 08:11 Sebastian Hildebrandt <notifications@github.com>
ha scritto:

**Comment by sebhildebrandt** [comment]
Funny enough - I just restarted my macbook and now it works fine ...

**Comment by borasemiz** [comment]
I don't have any antivirus software installed on my PC, however I can't enter the website either. I get the same SSL error message, including firebase.com too.

**Comment by mehdinh** [comment]
Same problem here on NOD32 Internet Security! 
Disabling real-time protection and firewall doesn't help. 
Uninstalled NOD32 and everything works while using Windows Defender. 

Really need to get this solved.
How is it possible that something like angular.io is blocked by an AV? Something smells!

**Comment by mbleigh** [comment]
Hi folks, we've just updated the SSL certificate used by angular.io to one that isn't shared by other Firebase Hosting customers. Can any of you who have been having problems check and see if you're having better luck now?

**Comment by ocombe** [comment]
It works for me, no more problem now !! :)

**Comment by mbleigh** [comment]
Woohoo!

**Comment by urbanhusky** [comment]
I don't know if I had issues at home, but it works here.
It will be a few days before I can test it at work - I'll report the results.

**Comment by mehdinh** [comment]
It works, thanks.

P.S.
I was playing with ESET Internet Security options. After changing SSL mode from automatic to interactive it worked. I reverted mode to automatic and it was still working. 
Exactly in that moment the cert was updated :).

**Comment by ShaneCourtrille** [comment]
Works behind a Fortigate firewall now!

Thanks very much!

**Comment by jeffbcross** [comment]
Well, it seems like the new cert worked! I'll close this issue now. If anyone experiences a similar issue, please create a new issue and mention me in it.

**Comment by jeffbcross** [comment]
And big thanks to @mbleigh for taking care of this!

**Comment by jlsjonas** [comment]
@mbleigh this does mean that all/most other firebase hosting customers are dealing with the same unreachability though...

**Comment by mbleigh** [comment]
We're working on a solution for everyone. Consider this a small test of
future progress 😉

**Comment by PaluMacil** [comment]
@jeffbcross @mbleigh My company cannot access the site due to our third party firewall declaring the cert bad. The initial October fix seemed to work, though problems recurred a couple weeks ago. I'm not sure I can help by opening a new issue because I won't be able to verify resolution (we can't do without the site any longer and will simply be whitelisting it by close of business today).

**Comment by PaluMacil** [comment]
I'm not sure why Github says I unassigned @jeffbcross

**Comment by mbleigh** [comment]
Dan: are you able to send me information about the certificate you're
seeing? Specifically the Common Name, Subject Alternate Names, and
expiration date?

**Comment by BLITZandKILL** [comment]
The issue is back for me as well, haven't visited angular.io in a while though so I don't know when it began again.

**Comment by ocombe** [comment]
No problem for me, it's still working

**Comment by mbleigh** [comment]
Actually, I'm seeing angular.io serving from a shared certificate again.
I'll investigate and report back.

**Comment by rodp82** [comment]
Stopped working for me as well. 
I've got Bitdefender running and turning off SSL scan allows me to access the site.
I've tried whitelisting the domain, but that didn't seem to help

**Comment by jeffswitzer** [comment]
Started experiencing the same symptoms again as @rodp82

**Comment by MoxxNull** [comment]
I am facing the same problems, but as rodp82 said, turning of ssl scanning in bitdefender does the trick.

**Comment by fredricadell** [comment]
For me in my company it complains that the TLS 1.0 to TLS 1.2 is deactivated. Probably for security reasons so the angular.* stack needs to be upped to the next level for us to be able to access the site.

**Comment by mbleigh** [comment]
We have an idea of what's causing this, but the fix is going to take a
little while. Please bear with us, and apologies for the inconvenience.

**Comment by markmaynard** [comment]
Any update on this? Still experiencing cert errors as of today.

**Comment by mattslay** [comment]
I am having this same exact problem using Chrome on Windows 10 Professional 64 bit.  I get basically the same error when trying it with Firefox.

> This site can’t provide a secure connection
> angular.io sent an invalid response.
> Try running Windows Network Diagnostics.
> ERR_SSL_PROTOCOL_ERROR

Chrome:
![2017-03-07_6-20-49](https://cloud.githubusercontent.com/assets/309016/23656204/50f55ebc-02fe-11e7-9656-17522da40f1f.png)

Firefox:
![2017-03-07_6-27-08](https://cloud.githubusercontent.com/assets/309016/23656330/eb0b26b2-02fe-11e7-9c9b-0a4c449b834b.png)

**Comment by Yolepo** [comment]
Hi,
Still experiencing this error too. I'm behind a stormshield firewall. Turning off the SSL validation does the work but this is not a good solution at all.

**Comment by mbleigh** [comment]
Hey all, can you give this another spin? We are no longer serving the same GlobalSign certificate and this may be resolved. Please let me know if you're still having problems!

**Comment by BLITZandKILL** [comment]
It seems that I can get to it just fine now.

**Comment by mxdumas** [comment]
Appears to be fixed with Bitdefender too. Working now. Thanks!

**Comment by mattslay** [comment]
Yes! It's working now. Thanks.

**Comment by urbanhusky** [comment]
Works fine with ESET, thanks :)

**Comment by fredricadell** [comment]
Works fine now! Thanks.

**Comment by Yolepo** [comment]
Works fine with à firewall. Thanks.
